File: cli_anything/hotelbyte/utils/output.py
"""utils/output.py — structured output helpers.

Handles the ``--json`` flag consistently: when set, emit compact JSON
to stdout for agent consumption; otherwise pretty-print or summarise
for humans.
"""
from __future__ import annotations

import json
import sys
from typing import Any

import click


def emit(data: Any, json_mode: bool, *, pretty: bool = True) -> None:
    """Emit ``data`` in the appropriate format.

    Args:
        data: the response payload (dict/list/str).
        json_mode: when True, emit ``--json`` structured output.
        pretty: when True (human mode), pretty-print with indentation.
    """
    if json_mode:
        click.echo(json.dumps(data, ensure_ascii=False, default=str))
    elif isinstance(data, (dict, list)):
        click.echo(json.dumps(data, ensure_ascii=False, indent=2 if pretty else None, default=str))
    elif data is None:
        click.echo("(no output)")
    else:
        click.echo(str(data))


def error(msg: str, json_mode: bool = False) -> None:
    """Emit an error message in the appropriate format."""
    if json_mode:
        click.echo(json.dumps({"error": msg}, ensure_ascii=False), err=True)
    else:
        click.echo(f"✗ {msg}", err=True)

File: src/utils/output.ts
/**
 * output.ts — structured output helpers.
 *
 * Handles the --json flag: when set, emit compact JSON to stdout for agent
 * consumption; otherwise pretty-print for humans.
 */

export function emit(data: unknown, jsonMode: boolean): void {
  if (jsonMode) {
    console.log(JSON.stringify(data));
  } else if (data === undefined || data === null) {
    console.log("(no output)");
  } else if (typeof data === "object") {
    console.log(JSON.stringify(data, null, 2));
  } else {
    console.log(String(data));
  }
}

export function error(msg: string, jsonMode = false): void {
  if (jsonMode) {
    console.error(JSON.stringify({ error: msg }));
  } else {
    console.error(`✗ ${msg}`);
  }
}

/**
 * Parse a JSON string or @file.json path into a JS object.
 */
export function parseJsonInput(value: string): unknown {
  if (value.startsWith("@")) {
    return JSON.parse(Bun.file(value.slice(1)).textSync());
  }
  try {
    return JSON.parse(value);
  } catch {
    return value; // keep as string
  }
}

/**
 * camelCase a kebab-case or snake_case option name.
 */
export function camelCase(key: string): string {
  return key.replace(/[-_]([a-z])/g, (_, c) => c.toUpperCase());
}